### PR TITLE
[geos] Update, cleanup

### LINKED ISSUE #24380: [geos] update to 3.10.2
Library name:geos

New version number:3.10.2

the current version has a problem with IsValid() test, this is a big problem, and the 3.10.2 has repaired it
https://libgeos.org/posts/2021-11-02-geos-3-10-2-released/

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Updates GEOS to 3.10.2. Resolves #24380.
  Cleanup and dropping of the debug postfix for easier mainenance. (AFAICT all downstream ports use pkg-config or cmake config. osgearth has a patch which refers to the geos debug postfix, but the patched file is unused, and osgearth deserves a separate cleanup.)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/geos/portfile.cmake [hunk 1]
```diff
@@ -1,24 +1,22 @@
...
     ARCHIVE "${ARCHIVE}"
-    REF ${GEOS_VERSION}
+    REF "${GEOS_VERSION}"
     PATCHES
         disable-warning-4996.patch
         fix-exported-config.patch
-        install-hpp-files.patch
 )
 
+set(EXTRA_OPTIONS)
```
Comment (JackBoosY) [inline ports/geos/portfile.cmake:17]:
```suggestion
set(EXTRA_OPTIONS "")
```

File: ports/geos/portfile.cmake [hunk 2]
```diff
@@ -29,36 +27,23 @@ vcpkg_cmake_configure(
...
-    if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_TARGET_IS_MINGW)
-        file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin")
-        file(RENAME "${CURRENT_PACKAGES_DIR}/bin/geos-config" "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin/geos-config")
-        file(CHMOD "${CURRENT_PACKAGES_DIR}/tools/${PORT}/bin/geos-config" FILE_PERMISSIONS
-            OWNER_READ OWNER_WRITE OWNER_EXECUTE
-            GROUP_READ GROUP_EXECUTE
-            WORLD_READ WORLD_EXECUTE
-        )
-    endif()
-endif()
```
Comment (JackBoosY) [inline ports/geos/portfile.cmake:55]: Is this not needed?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/geos/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/geos/portfile.cmake:17]: Indeed, it was meant like this.

File: ports/geos/portfile.cmake (hunk 2 above)
Comment (dg0yt) [reply ports/geos/portfile.cmake:55]: The debug postfix is removed in this PR so it is no longer needed.
The `geos-config` fixup for release is still there, but no longer guarded by `VCPKG_BUILD_TYPE`. (Debug-only single-config is unsupported.)

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the fixes!

**Pull request merged**